### PR TITLE
Add ANCM stress test app

### PR DESCRIPTION
**Pull request by bangbingsyb**
Working on developing stress tests for IIS ASP.NET Core Module and IISIntegration. One of the challenge for the daily stress test is that we want to build the test app with all the dependencies up-to-date. 

The proposed approach is to add the app as a sample to ServerTests repo, and update the PackageReference in the project file to consume the version variables provided by the repo.

### CODE REVIEW

File: samples/ANCMStressTestSample/Startup.cs [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+            bool closeFromServer = false;
+            string closeFromServerCmd = "CloseFromServer";
+            int closeFromServerLength = closeFromServerCmd.Length;
+
+            bool echoBack = true;
+            int repeatCount = 1;
+
+            while (!result.CloseStatus.HasValue)
+            {
+                if ((result.Count == closeFromServerLength && System.Text.Encoding.ASCII.GetString(buffer).Substring(0, result.Count) == closeFromServerCmd)
```
Comment (jkotalik) [inline samples/ANCMStressTestSample/Startup.cs]: I don't feel like transmitting data across the websocket for commands is appropriate. I think these should be transmitted via Headers on the upgrade request test one individual component per websocket request.
cc @anurse

File: samples/ANCMStressTestSample/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+                        {
+                            await webSocket.SendAsync(new ArraySegment<byte>(buffer, 0, result.Count), result.MessageType, result.EndOfMessage, CancellationToken.None);
+                        }
+                    }
+                }
+
+                result = await webSocket.ReceiveAsync(new ArraySegment<byte>(buffer), CancellationToken.None);
+            }
+
+            if (closeFromServer)
```
Comment (jkotalik) [inline samples/ANCMStressTestSample/Startup.cs]: Shouldn't this be closeFromClient?

**Comment by Tratcher** [comment]
Let's move this back to IISIntegration. The project is almost an exact duplicate of the one already there. As for WebSockets it can be handled one of two ways:
1) Write a WebSocket specific test in the WebSocket repo. We should do this for Autobahn anyways.
2) Don't depend on the middleware, most of the functionality comes directly from CoreFx so you can call it directly. You primarily need these lines:
https://github.com/aspnet/WebSockets/blob/6de76c5e69bd74b8d5742642e1e90a2a39874323/src/Microsoft.AspNetCore.WebSockets/WebSocketMiddleware.cs#L42
https://github.com/aspnet/WebSockets/blob/6de76c5e69bd74b8d5742642e1e90a2a39874323/src/Microsoft.AspNetCore.WebSockets/WebSocketMiddleware.cs#L120-L125

**Comment by bangbingsyb** [comment]
@Tratcher Yeah, pretty much a replication of the sample app there + websocket. I'd prefer IISIntegration as well to make everything needed from one place. And the whole reason for bothering ServerTest is because this circular dependency issue of websocket. Let me try to directly call CoreFx in the app.

**Comment by jkotalik** [comment]
@bangbingsyb ping me if you need any help modifying the IISIntegration app.

**Comment by bangbingsyb** [comment]
@Tratcher @jkotalik Go back to the option of removing dependency on the WebSockets middleware. I briefly reviewed the middleware source, and want to confirm with you whether the proposed change is the right thing to do:

1. For IHttpWebSocketFeature, looks like I need to duplicate the entire private class implementation UpgradeHandshake from WebSocket repo. Then simply instantiate it and set it to context.Features. 
2. For UpgradeHandshake class, I also need to duplicate other helper classes including HandshakeHelpers, Constants, ExtendedWebSocketAcceptContext.

I feel if that's the case, I'm duplicating half of the WebSockets middleware code.

**Comment by Tratcher** [comment]
You don't need the IHttpWebSocketFeature at all. You can bypass it and go directly to the IHttpUpgradeFeature feature. Most of IHttpWebSocketFeature is argument validation anyways.  Let me know when you have a few minutes and we can hack together a sample in person.

**Comment by bangbingsyb** [comment]
Since I've sent the PR to IISIntegration, close this PR.

**Pull request closed without merging**